controllers/commonservice_controller.go

```diff
@@ -96,22 +128,52 @@ func (r *CommonServiceReconciler) ReconcileMasterCR(req ctrl.Request) (ctrl.Resu
 // ReconcileGeneralCR is for setting the OperandConfig
 func (r *CommonServiceReconciler) ReconcileGeneralCR(req ctrl.Request) (ctrl.Result, error) {
 
+	// Fetch the CommonService instance
+	instance := &apiv3.CommonService{}
+
+	if err := r.Client.Get(ctx, req.NamespacedName, instance); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if instance.Status.Phase == "" {
+		if err := r.updatePhase(instance, CRInitializing); err != nil {
+			return ctrl.Result{}, err
+		}
+	} else {
+		if err := r.updatePhase(instance, CRUpdating); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
 	opcon := util.NewUnstructured("operator.ibm.com", "OperandConfig", "v1alpha1")
 	opconKey := types.NamespacedName{
 		Name:      "common-service",
 		Namespace: constant.MasterNamespace,
 	}
 	if err := r.Reader.Get(ctx, opconKey, opcon); err != nil {
 		klog.Errorf("failed to get OperandConfig %s: %v", opconKey.String(), err)
+		if err := r.updatePhase(instance, CRFailed); err != nil {
+			klog.Error(err)
+		}
 		return ctrl.Result{}, err
 	}
 
 	newConfigs, err := r.getNewConfigs(req)
 	if err != nil {
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+		if err := r.updatePhase(instance, CRFailed); err != nil {
+			klog.Error(err)
+		}
+		return ctrl.Result{}, err
 	}
 
 	if err = r.updateOperandConfig(newConfigs); err != nil {
+		if err := r.updatePhase(instance, CRFailed); err != nil {
+			klog.Error(err)
+		}
+		return ctrl.Result{}, err
+	}
```

Comment:
This is the same as above one.

Would we expect this situation since the first err is not reported in log?